test/levels.test.js

```diff
@@ -0,0 +1,92 @@
+import _ from 'lodash';
+import levels from '../src/levels';
+
+describe('levels', () => {
+  it('should be defined', () => {
+    let expectedLevels = [
+      'alert',
+      'critical',
+      'debug',
+      'emergency',
+      'error',
+      'fatal',
+      'info',
+      'informational',
+      'notice',
+      'silly',
+      'trace',
+      'verbose',
+      'warn',
+      'warning'];
+
+    _.forEach(expectedLevels, (lvl) => {
+      expect(levels[lvl]).toBeDefined();
+    });
+  });
+});
+
+describe('levelToLevelName', () => {
+  it('should be defined', () => {
+    expect(levels.levelToLevelName).not.toBeUndefined();
+  });
+
+  it('should return as expected for defined levels', () => {
+    expect(levels.levelToLevelName(levels.alert)).toBe('alert');
+    expect(levels.levelToLevelName(levels.critical)).toBe('critical');
+    expect(levels.levelToLevelName(levels.debug)).toBe('debug');
+    expect(levels.levelToLevelName(levels.emergency)).toBe('fatal');
+    expect(levels.levelToLevelName(levels.error)).toBe('error');
+    expect(levels.levelToLevelName(levels.fatal)).toBe('fatal');
+    expect(levels.levelToLevelName(levels.info)).toBe('info');
+    expect(levels.levelToLevelName(levels.informational)).toBe('info');
+    expect(levels.levelToLevelName(levels.notice)).toBe('notice');
+    expect(levels.levelToLevelName(levels.silly)).toBe('silly');
+    expect(levels.levelToLevelName(levels.trace)).toBe('trace');
+    expect(levels.levelToLevelName(levels.verbose)).toBe('debug');
+    expect(levels.levelToLevelName(levels.warn)).toBe('warn');
+    expect(levels.levelToLevelName(levels.warning)).toBe('warn');
```

Comment:
levels (or defaultLevels i.e. without the functions inside) is just a key-value of name to number.
once that is corrected, you'd loop over and the this in a generic way

Reply:
re:expected above
if you want to create regression/snapshot tests, you have jest's snapshots feature for that
https://facebook.github.io/jest/docs/en/snapshot-testing.html

just as an exercise (don't commit it :) ), now that you loop over the levels, instead of having an expect inside the loop, you could record the input and output of levelToLevelName in a variable and say "expect(current).toMatchSnapshot()";

I'm saying don't commit it, because you'd test the wrong thing. If we add another level, the snapshot test would fail, though the levelToLevelName might actually do the right thing.